### PR TITLE
#161254662 Create user profile upon registration

### PR DESCRIPTION
**Pull request by TheSteelGuy**
#### What does this PR do?
```
- creates a profile once a user registers
- a user can retrieve his/her profile
- a user can retrieve all users profiles
- implement update user profile

```
#### Description of Task to be completed?
```
User profile should be automatically created once the user registers
Users who are authenticated should be able to view other profiles but not edit them
An authenticated user should be able to view another user's profile but not edit it
 ```
#### How should this be manually tested?

- create a python virtual environment in your local machine and start it.
```
$ virtualenv env
$ source env/bin/activate
$ cd env
```
- Clone this repo.
``` 
$ git clone https://github.com/andela/ah-orcas.git
```
- Switch to this branch
``` 
$  git checkout  ft-user-profile-16125466
```
- Create a .env file and run it. Check sample.env file to configure yours.
```
$ source .env
```
- Run the app
```
$ python manage.py runserver
```
- Test the endpoint as per the README.md

**Any background context you want to provide?**

```All endpoints require authentication i.e jwt token```

**What are the relevant pivotal tracker stories?**
[#161254662](https://www.pivotaltracker.com/story/show/161254662)

### CODE REVIEW

File: authors/apps/profiles/serializers.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from .models import UserProfile
+from rest_framework import serializers
+
+image = 'http://correctimage.png'
```
Comment (code-sleuth) [inline authors/apps/profiles/serializers.py]: Why are we using this?

File: authors/apps/articles/tests/test_rate_article.py [hunk 2]
```diff
@@ -65,9 +65,9 @@ def setUp(self):
                 "rate": 3
             }
         }
-        self.rate_url = os.environ["URL"]+"api/article/"\
+        self.rate_url = "http://localhost:8000/"+"api/article/"\
```
[the remark this answers is not in the document]
Comment (TheSteelGuy) [reply authors/apps/articles/tests/test_rate_article.py]: Check...

File: authors/apps/authentication/models.py [hunk 3]
```diff
@@ -87,11 +87,11 @@ class User(AbstractBaseUser, PermissionsMixin):
     # at signup let the user confirmed status be false
     is_confirmed_email = models.BooleanField(default=False)
 
-    # A timestamp representing when this object was created.
-    created_at = models.DateTimeField(auto_now_add=True)
-
-    # A timestamp reprensenting when this object was last updated.
-    updated_at = models.DateTimeField(auto_now=True)
+    # # A timestamp representing when this object was created.
```
[the remark this answers is not in the document]
Comment (TheSteelGuy) [reply authors/apps/authentication/models.py]: check..

File: authors/apps/profiles/serializers.py (hunk 1 above)
Comment (TheSteelGuy) [reply authors/apps/profiles/serializers.py]: We wanted to have like a default avatar, but pylint kept complaining that it was long, but we have removed that part

File: authors/apps/core/exceptions.py [hunk 4]
```diff
@@ -23,6 +23,7 @@ def core_exception_handler(exc, context):
 
     return response
 
+
```
Comment (kipropbrian) [inline authors/apps/core/exceptions.py]: There is not change here. So it is unnecessary

File: authors/apps/profiles/tests/test_user_profile.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    def test_get_all_profiles(self):
+        """
+        test retrieve all the available profiles
+        """
+        self.register_user()
+        token = self.login_user()
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + token)
+        response = self.client.get(self.all_profiles)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn('user1', str(response.data))
```
Comment (kipropbrian) [inline authors/apps/profiles/tests/test_user_profile.py]: Could you also do a count of all profiles and see if they match.

File: authors/apps/profiles/views.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+            )
+
+
+class ProfileListAPIView(ListAPIView):
+
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+
+    def get(self, request, **kwargs):
+        """
+        Fetch the all the available profiles
```
Comment (kipropbrian) [inline authors/apps/profiles/views.py]: Grammar issues here

File: authors/apps/profiles/tests/test_user_profile.py (hunk 5 above)
Comment (TheSteelGuy) [reply authors/apps/profiles/tests/test_user_profile.py]: on it

File: authors/apps/profiles/views.py (hunk 6 above)
Comment (TheSteelGuy) [reply authors/apps/profiles/views.py]: 👍

File: authors/apps/core/exceptions.py (hunk 4 above)
Comment (wrotich) [reply authors/apps/core/exceptions.py]: pep8 requires 2 extra lines, it found one so we added the other one

**Pull request merged**